server/controllers/survey.ts

```diff
@@ -61,12 +61,21 @@ export function displayQuestionPage(req:Request, res: Response, next: NextFuncti
 export function displayEditSurveyPage(req:Request, res: Response, next: NextFunction):void
 {
     const id = req.params.id;
-
-    getSurveyById(id, (err, survey) => {
-        if (err) {
+    if (!req.user) { // if req.user is undefined
+        throw Error("Unreachable: this route handler is called only when the user is logged in");
+    }
+    const userId = req.user._id;
+    //survey.find() by using  survey id and owner(user id) to make sure the user edits own survey
+    Survey.find({ owner:userId, _id:id},(err)=> {
+        if(err){
             return next(err);
         }
-        res.render("index", { title: "EditSurvey", page: "editsurvey", surveyItem: survey });
+        getSurveyById(id, (err,survey) => {
```

Comment:
You are making a `find` call on the database twice to get the same object. We do not want this to minimize database calls. I would suggest using `getSurveyById` only (and get rid of the `Survey.find` call above) and then checking if the `userId` is the same as the `owner` field of the survey object.

```ts
getSurveyById(id, (err, survey) => {
    if (err) {
        return next(err);
    }

    if (!userId.equals(survey.owner)) {
        return res.redirect("/account"); // redirect to the account page
        // OR you could also try sending out 401 Unauthorized error (more advanced)
    }

    // Now you can render the page here
});
```